subprojects/core/src/main/java/org/gradle/internal/typeconversion/NotationParserBuilder.java

```diff
@@ -134,4 +134,17 @@ private NotationParserBuilder(TypeInfo<T> resultingType) {
 
         return new NotationConverterToNotationParserAdapter<Object, T>(composites.size() == 1 ? composites.get(0) : new CompositeNotationConverter<Object, T>(composites));
     }
+
+    private static class LazyDisplayName<T> {
```

Comment:
We want a concept we can reuse everywhere, but not necessarily an implementation. In other words, we might have an interface something like:

```interface Describable {
    String getDisplayName()
}```

And then incrementally have more things provide and understand `Describable` (or `HasDescriptor` or whatever).